src/main/java/com/ericsson/gerrit/plugins/eiffel/listeners/AbstractEventListener.java

```diff
@@ -0,0 +1,109 @@
+import com.google.gerrit.server.config.CanonicalWebUrl;
+import com.google.gerrit.server.events.ChangeEvent;
+import com.google.gerrit.server.events.Event;
+import com.google.inject.Inject;
+
+public abstract class AbstractEventListener implements EventListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ChangeMergedEventListener.class);
+
+    @Inject
+    private com.google.gerrit.server.config.PluginConfigFactory pluginConfigFactory;
+
+    @Inject
+    @CanonicalWebUrl
+    private String gerritUrl;
+
+    protected final String pluginName;
+    protected final File plugindir;
+
+    public AbstractEventListener(final String pluginName, final File pluginDir) {
+        this.pluginName = pluginName;
+        this.plugindir = pluginDir;
+    }
+
+    public boolean isEventSendingEnabled(final Event gerritEvent,
+            final EiffelPluginConfiguration pluginConfig) {
+
+        if (!isPluginEnabled(pluginConfig)) {
+            return false;
+        }
+
+        ChangeEvent changeEvent = (ChangeEvent) gerritEvent;
+        final String project = changeEvent.change.get().project;
+        final String branch = changeEvent.change.get().branch;
+        final String filter = pluginConfig.getFilter();
+        if (isFilteredOut(branch, filter, project)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected abstract boolean isExpectedGerritEvent(Event gerritEvent);
+
+    protected abstract void prepareAndSendEiffelEvent(Event gerritEvent,
+            EiffelPluginConfiguration pluginConfig);
+
+    protected EiffelPluginConfiguration createPluginConfig(Event gerritEvent) {
+        ChangeEvent changeEvent = (ChangeEvent) gerritEvent;
+        final Project.NameKey projectNameKey = changeEvent.getProjectNameKey();
+        final EiffelPluginConfiguration pluginConfig = new EiffelPluginConfiguration(pluginName,
+                projectNameKey, pluginConfigFactory);
+        return pluginConfig;
+    }
+
+    private boolean isPluginEnabled(EiffelPluginConfiguration pluginConfig) {
+        if (!pluginConfig.isEnabled()) {
+            LOGGER.debug("Eiffel plugin is disabled for this project.\n"
+                    + "Please refer to Eiffel plugin documentation to find out how to configure and enable plugin\n"
+                    + "{}plugins/{}/Documentation/index.html", gerritUrl, pluginName);
+            return false;
+        }
+
+        return true;
+    }
+
+    private boolean isFilteredOut(String branch, String filter, String project) {
+        boolean branchIsFilteredOut = true;
+        if (filter == null || filter.isEmpty()) {
+            branchIsFilteredOut = false;
+        }
+
+        for (final String regExString : filter.split("\\s+")) {
+            if (branch.matches(regExString)) {
+                branchIsFilteredOut = false;
+            }
+        }
+
+        LOGGER.debug("Branch {} for project {} doesn't much any of filters '{}', skip sending",
```

Comment:
"much any of filters" ,, you mean probably , "match any filters" or "match any of the configured filters"

Reply:
Done in next commit.